src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -188,9 +191,20 @@ public static function createConnection(#[\SensitiveParameter] string $dsn, arra
             $params['lazy'] = filter_var($params['lazy'], \FILTER_VALIDATE_BOOLEAN);
         }
         $params['redis_cluster'] = filter_var($params['redis_cluster'], \FILTER_VALIDATE_BOOLEAN);
-
-        if ($params['redis_cluster'] && isset($params['redis_sentinel'])) {
-            throw new InvalidArgumentException('Cannot use both "redis_cluster" and "redis_sentinel" at the same time.');
+        $params['relay_cluster'] = filter_var($params['relay_cluster'], \FILTER_VALIDATE_BOOLEAN);
+
+        $conflictingOptions = array_filter([
+            'redis_cluster'  => $params['redis_cluster'],
+            'relay_cluster'  => $params['relay_cluster'],
```

Comment:
Why do we need a `relay_cluster` option ? Shouldn't this be based on `redis_cluster` with the relay extension loaded, just like other cases switching between ext-redis, ext-relay and predis ?

`redis_sentinel` enables the sentinel mode for any of the 3 supported backends. and `redis_cluster` currently enables it for ext-redis and predis already.

Reply:
switching automatically between extensions is exactly what happens for the non-cluster case

Reply:
That is true - and I suppose the consumer of the library can always pass in his own instance if he wants to instead of relying on the connection string approach. Let me update it so that it is automatic in the same way then.